sklearn/preprocessing/_data.py

```diff
@@ -2059,9 +2066,10 @@ def binarize(X, *, threshold=0.0, copy=True):
         Threshold may not be less than 0 for operations on sparse matrices.
 
     copy : bool, default=True
-        Set to False to perform inplace binarization and avoid a copy
-        (if the input is already a numpy array or a scipy.sparse CSR / CSC
-        matrix and if axis is 1).
+        If False, try to avoid a copy and scale in place.
+        This is not guaranteed to always work in place; e.g. if the data is
```

Comment:
I think in this case (and maybe others) the previous wording is accurate that scipy CSR or CSC matrices are not copied, so you can not reuse the same doc here.

In general, the fact that a copy is made mostly depends on the `check_array` arguments, but the code following it may also need to be looked at closer.

Reply:
I still think the previous docstring was accurate whether the new one is slightly misleading.

The code below is used, which means that it accepts CSR/CSC sparse matrix without copying it:
```py
X = check_array(X, accept_sparse=["csr", "csc"], copy=copy)
```